tests/functional/db_cleanup/test_db_cleanup.py

```diff
@@ -0,0 +1,159 @@
+#
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+import json
+import multiprocessing
+import os
+import unittest
+import time
+
+from codeCheckerDBAccess_v6.ttypes import *
+
+from libtest import codechecker
+from libtest import env
+
+
+class TestDbCleanup(unittest.TestCase):
+
+    def setUp(self):
+        self.test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self.test_workspace)
+
+        self.codechecker_cfg = env.import_codechecker_cfg(self.test_workspace)
+        self.test_dir = os.path.join(self.test_workspace, 'test_files')
+
+        try:
+            os.makedirs(self.test_dir)
+        except os.error:
+            # Directory already exists.
+            pass
+
+    def _create_test_dir(self):
+        makefile = "all:\n\t$(CXX) -c a/main.cpp -o /dev/null\n"
+        project_info = {
+            "name": "hello",
+            "clean_cmd": "",
+            "build_cmd": "make"
+        }
+        source_main = """
```

Comment:
Why are we embedding these sources in the python script and write them out? Wouldn't it be better to keep it simple and just check in those files to the repository?

Reply:
Later in the test case these files are replaced to another path, that's why these test files are not committed to the repository. There are other test cases which use this technique when test files change in any way.